components/Meta.tsx

```diff
@@ -0,0 +1,75 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+import { SITE_DESCRIPTION, SITE_NAME } from "@/constants.ts";
+
+interface MetaProps {
+  title?: string;
+  description?: string;
+  imageUrl?: string;
+  url?: string;
+}
+
+export default function Meta(props: MetaProps) {
+  const { title, description, imageUrl, url } = props;
+
+  return (
+    <>
+      {/* HTML Meta Tags */}
+      <title>{title || SITE_NAME}</title>
+      <meta
+        name="description"
+        content={description || SITE_DESCRIPTION}
+      />
+      <meta property="og:type" content="website" />
+      <meta property="og:site_name" content={title || SITE_NAME} />
+      <meta property="og:locale" content="en" />
+
+      {/* Google / Search Engine Tags */}
+      <meta
+        itemProp="name"
+        content={title || SITE_NAME}
+      />
+      <meta
+        itemProp="description"
+        content={description || SITE_DESCRIPTION}
+      />
+      <meta
+        itemProp="image"
+        content={imageUrl}
+      />
+
+      {/* Facebook Meta Tags */}
+      <meta
+        property="og:url"
+        content={url}
+      />
+      <meta property="og:type" content="website" />
+      <meta
+        property="og:title"
+        content={title || SITE_NAME}
+      />
+      <meta
+        property="og:description"
+        content={description || SITE_DESCRIPTION}
+      />
+      <meta
+        property="og:image"
+        content={imageUrl}
+      />
```

Comment:
`imageUrl` is optional. I think it's best to only generate this and other `imageUrl`-containing elements, with a conditional that checks for the existence of `imageUrl` first. E.g. `{imageUrl && <Element>}`.

This behaviour should also occur for other properties of `MetaProps` that lack a default value.

Reply:
That's what I also though about. Did not know how Twitter etc. reacts when there is a tag without a value. Definitely makes sense to render these conditionally.